agent/canvas.py

```
@@ -101,20 +101,28 @@ def __init__(self, dsl: str, tenant_id=None, task_id=None, custom_header=None):
+        self.path = self.dsl["path"]
+
+    @staticmethod
+    def validate_component_parameters(dsl):
+        component_params = {}
+        for k, cpn in dsl["components"].items():
             param = component_class(cpn["obj"]["component_name"] + "Param")()
-            cpn["obj"]["params"]["custom_header"] = self.custom_header
             param.update(cpn["obj"]["params"])
             try:
                 param.check()
             except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)